sql/rowexec/ddl.go

```diff
@@ -511,9 +513,8 @@ func (b *BaseBuilder) buildCreateUser(ctx *sql.Context, n *plan.CreateUser, row
 }
 
 func (b *BaseBuilder) buildAlterPK(ctx *sql.Context, n *plan.AlterPK, row sql.Row) (sql.RowIter, error) {
-	// We grab the table from the database to ensure that state is properly refreshed, thereby preventing multiple keys
-	// being defined.
-	// Grab the table fresh from the database.
+	// We need to get the current table from the database because this statement could be one clause in an alter table
+	// statement and the table may have changed since the analysis phase
```

Comment:
seems like we should could have a different type of execution structure for alter tables. Doesn't really lend itself to volcano.

Reply:
Maybe, work for another day